contracts/examples/BalanceCredentialIssuer.sol

```diff
@@ -1,85 +1,187 @@
     uint64 private countOfIssuedClaims = 0;
     mapping(uint256 => Claim[]) private claimStorage;
 
     uint256[48] private __gap_after;
 
-    function initialize(
-        address _stateContractAddr
-    ) public initializer override {
+    function initialize(address _stateContractAddr) public override initializer {
         IdentityBase.initialize(_stateContractAddr);
+        state = IState(_stateContractAddr);
         __Ownable_init();
     }
 
     /**
-     * @dev Get schema info by schema hash
-     * @param _jsonldSchemaHash - schema hash
+     * @dev Get user's credentials ids
+     * @param _userId - user id
      */
-    function getSchemaInfo(uint256 _jsonldSchemaHash) public pure returns (SchemaInfo memory) {
-        require(_jsonldSchemaHash == jsonldSchemaHash, "Invalid schema hash");
-        return SchemaInfo({
-            jsonldSchemaURL: "https://gist.githubusercontent.com/ilya-korotya/ac20f870943abd4805fe882ae8f3dccd/raw/1d9969a6d0454280c8d5e79b959faf9b3978b497/balance.jsonld",
-            jsonSchemaURL: "https://gist.githubusercontent.com/ilya-korotya/26ba81feb4da2f49f4b473661b80e8e3/raw/32113f4725088f32f31a6b06b4abdc94bc4b2d17/balance.json",
-            credentialType: "Balance"
-        });
+    function listUserCredentials(
+        uint256 _userId
+    ) public view returns (IW3CVerifiableCredential.Id[] memory) {
+        Claim[] memory claims = claimStorage[_userId];
+        IW3CVerifiableCredential.Id[] memory credentials = new IW3CVerifiableCredential.Id[](
+            claims.length
+        );
+        for (uint i = 0; i < claims.length; i++) {
+            credentials[i] = IW3CVerifiableCredential.Id(
+                claims[i].credentialMetadata.sequenceNumber
+            );
+        }
+        return credentials;
     }
 
     /**
-     * @dev Get all user claims by user id
+     * @dev Get user's verifiable credentials
      * @param _userId - user id
+     * @param _credentialId - credential id
      */
-    function getUserClaims(uint256 _userId) public view returns (Claim[] memory) {
-        return claimStorage[_userId];
+    function getCredential(
+        uint256 _userId,
+        uint256 _credentialId
+    ) public view returns (IW3CVerifiableCredential.Credential memory) {
+        Claim[] memory claims = claimStorage[_userId];
+        for (uint i = 0; i < claims.length; i++) {
+            if (claims[i].credentialMetadata.sequenceNumber == _credentialId) {
+                return convertToVerifiableCredential(claims[i]);
+            }
+        }
+        revert('Credential not found');
     }
 
-    /**
-     * @dev Get user claim by user id and index
-     * @param _userId - user id
-     * @param _index - index of the claim
-     */
-    function getUserClaimByIndex(uint256 _userId, uint _index) public view returns (Claim memory) {
-        return claimStorage[_userId][_index];
+    function convertToVerifiableCredential(
```

Comment:
I'm not sure about it. Since we can't have a common code for credential subject or credential status, these can be different for different users and contracts.